jmslib/src/main/java/com/redhat/mqe/lib/CoreClient.java

```diff
@@ -336,6 +337,11 @@ protected void close(Session session) {
         try {
             LOG.trace("Closing session " + session.toString());
             session.close();
+        } catch (JmsConnectionRemotelyClosedException rce) {
+            LOG.warn("JMS Connection Remotely Closed");
+            // wrt the current cli-* listener clients implementation,
+            // the connection close from remote peer might be expected
+            System.exit(0);
```

Comment:
Can you do something like this, in the same file above this function? What happens if you leave out `System.exit(0)`? Wont the client exit cleanly on its own then?

```
} catch (JMSException e) {
            e.printStackTrace();
            e.printStackTrace();
            if (e.getCause() instanceof SocketException
            if (e.getCause() instanceof SocketException
                && e.getCause().getMessage().equals("Connection closed by remote host")) {
                && e.getCause().getMessage().equals("Connection closed by remote host")) {
                return;  // suppress error, explained at https://issues.apache.org/jira/browse/AMQ-6956
                return;  // suppress error, explained at https://issues.apache.org/jira/browse/AMQ-6956
            }
            }
            System.exit(1);
            System.exit(1);
        }
        }
```

Reply:
you're right, it exits cleanly. That JmsConnectionRemotelyClosedException catch statement (which was added) ensures the exception won't be caught as the JMSException below which exits with 1. So, we don't need to explicitly sys exit with 0.